code/controllers/subsystem/job.dm

```diff
@@ -402,6 +402,10 @@ SUBSYSTEM_DEF(job)
 			else
 				M = H
 
+	var/ANTAG_REP_MAXIMUM = CONFIG_GET(number/antag_rep_maximum)
+	SSpersistence.antag_rep[M.client.ckey] = min(SSpersistence.antag_rep[M.client.ckey]+job.antag_rep, ANTAG_REP_MAXIMUM)
```

Comment:
This would only work for latejoins, at this point during roundstart keys/clients are not transferred yet.

Reply:
I think it would also rep people before antags are rolled, which is not intended. You should only receive rep if you're playing a job that gives it and you're not an antag.

Where's a more appropriate place to put this?

Reply:
This is alright place for this. You can use special_role for antag check , since this is still part of roundstart assignment.

Reply:
There is however the problem of late joining antags, which get assigned after equipment without setting special role for now. Would need some changes to that system for this to work properly.

Reply:
has this feedback been addressed?